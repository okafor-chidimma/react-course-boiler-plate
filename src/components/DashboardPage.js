import React from "react";

const DashBoardPage = () => (
  <div>
    Dashboard content page
  </div>
);

export default DashBoardPage;
