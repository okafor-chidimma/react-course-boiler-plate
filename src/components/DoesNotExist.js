import React from 'react';
import { Link } from 'react-router-dom';

const DoesNotExist = () => (
  <div>
    404page!<Link to="/">Home</Link>
  </div>
);

export default DoesNotExist;