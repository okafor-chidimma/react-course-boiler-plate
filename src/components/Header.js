import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogOutProcess } from '../actions/auth';

export const Header = ({ startLogout }) => (
  <header className='header'>
    <div className='content-container'>
      <div className="header__content">
        <Link className='header__title' to='/dashboard'>
          <h1>Boiler Plate</h1>
        </Link>
        <button onClick={startLogout} className="button button--link">Log out</button>
      </div>
    </div>

    {
      // <NavLink to="/edit/:id" activeClassName="is-active">Edit</NavLink>
      // <NavLink to='/help' activeClassName='is-active'>
      //   Help
      // </NavLink>
      //  <NavLink to='/create' activeClassName='is-active'>
      //    Create Expense
      //  </NavLink>
    }
  </header>
);

const matchDispatchToState = dispatch => ({
  startLogout: () => dispatch(startLogOutProcess())
});
export default connect(undefined, matchDispatchToState)(Header);
