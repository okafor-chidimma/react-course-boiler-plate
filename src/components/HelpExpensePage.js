import React from 'react';
const HelpExpensePage = () => (
  <div>
    this is the help expense page
  </div>
);

export default HelpExpensePage;