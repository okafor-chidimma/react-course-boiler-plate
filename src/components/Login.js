import React from 'react';
import { connect } from 'react-redux';
import { startLoginProcess } from '../actions/auth';
export const Login = ({ startLogin }) => (
  <div className='box-layout'>
    <div className='box-layout__box'>
      <h1 className='box-layout__title'>Boiler Plate</h1>
      <p>Tag Line For Application</p>
      <button className='button' onClick={startLogin}>
        Login With Google
      </button>
    </div>
  </div>
);

// since we want to dispatch the startLoginProcess() to fire up th authentication
const mapDispatchToState = dispatch => {
  return {
    startLogin: () => dispatch(startLoginProcess())
  };
};

export default connect(undefined, mapDispatchToState)(Login);
