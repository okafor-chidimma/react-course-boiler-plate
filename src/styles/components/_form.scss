.form {
  display: flex;
  flex-direction: column;
  > * {
    margin-bottom: $m-size;
  }
  .SingleDatePickerInput__withBorder_2 {
    width: 100% !important;
  }
}

.form__error {
  margin: 0 0 $m-size 0;
  font-style: italic;
}
