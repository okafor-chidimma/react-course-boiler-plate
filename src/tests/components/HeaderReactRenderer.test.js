import React from 'react';
import ReactShallowRenderer from 'react-test-renderer/shallow';
import { Header } from '../../components/Header';

describe('To test the Header component with react renderer package', () => {
  test('should render Header correctly', () => {
    const renderer = new ReactShallowRenderer();
    renderer.render(<Header />);
    // console.log(renderer.getRenderOutput());
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
