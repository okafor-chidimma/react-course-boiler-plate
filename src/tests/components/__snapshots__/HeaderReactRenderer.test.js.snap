// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`To test the Header component with react renderer package should render Header correctly 1`] = `
<header
  className="header"
>
  <div
    className="content-container"
  >
    <div
      className="header__content"
    >
      <ForwardRef
        className="header__title"
        to="/dashboard"
      >
        <h1>
          Boiler Plate
        </h1>
      </ForwardRef>
      <button
        className="button button--link"
      >
        Log out
      </button>
    </div>
  </div>
</header>
`;
